pkg/cmd/authenticated_cli_command.go

```diff
@@ -296,12 +370,67 @@ func (c *AuthenticatedCLICommand) GetCurrentSchemaRegistryClusterIdAndEndpoint()
 	}
 	cluster := clusters[0]
 	var endpoint string
-
-	if cluster.Spec.GetHttpEndpoint() != "" {
-		endpoint = cluster.Spec.GetHttpEndpoint()
+	if c.Context.GetSchemaRegistryEndpoint() != "" {
+		endpoint = c.Context.GetSchemaRegistryEndpoint()
 	} else {
-		endpoint = cluster.Spec.GetPrivateHttpEndpoint()
+		endpoint, err = c.GetValidSchemaRegistryClusterIdAndEndpoint(cmd, cluster)
+		if err != nil {
+			return "", "", err
+		}
 	}
-
 	return cluster.GetId(), endpoint, nil
 }
+
+func (c *AuthenticatedCLICommand) GetValidSchemaRegistryClusterIdAndEndpoint(cmd *cobra.Command, cluster srcmv3.SrcmV3Cluster) (string, error) {
+	unsafeTrace, _ := cmd.Flags().GetBool("unsafe-trace")
+	configuration := srsdk.NewConfiguration()
+	configuration.UserAgent = c.Config.Version.UserAgent
+	configuration.Debug = unsafeTrace
+	configuration.HTTPClient = ccloudv2.NewRetryableHttpClient(c.Config, unsafeTrace)
+
+	schemaRegistryApiKey, _ := cmd.Flags().GetString("schema-registry-api-key")
+	schemaRegistryApiSecret, _ := cmd.Flags().GetString("schema-registry-api-secret")
+	isValid := false
```

Comment:
I think we should rename this, but I don't have any suggestion at the moment.

In particular, on-prem depending on `isValid` being false in the last part of the function makes the name a bit unintuitive.